finetune/target_models/sequence_labeling.py

```diff
@@ -193,6 +205,25 @@ def _initialize(self):
         self.multi_label = self.config.multi_label_sequences
         return super()._initialize()
 
+    def finetune(self, Xs, Y=None, context=None, update_hook=None):
+        if self.config.auto_negative_sampling and Y is not None:
+            model_copy = copy.deepcopy(self)
+            model_copy._initialize()
+            model_copy.input_pipeline.total_epoch_offset = self.config.n_epochs
+            self.input_pipeline.current_epoch_offset = self.config.n_epochs
+            self.input_pipeline.total_epoch_offset = self.config.n_epochs
+
+            model_copy.config.max_empty_chunk_ratio = 0.0
+            model_copy.config.auto_negative_sampling = False
+            # TODO: fix update hooks progress
+            model_copy.finetune(Xs, Y=Y, context=context, update_hook=update_hook)
+            initial_run_preds = model_copy.predict(Xs)
+            del model_copy
+            Y_with_neg_samples = negative_samples(initial_run_preds, Y, pad=self.config.pad_token)
+            self.config.max_empty_chunk_ratio *= sum(len(yi) for yi in Y) / sum(len(yi) for yi in Y_with_neg_samples)
```

Comment:
Heuristic here is -- ratio of positive labels to FP labels should be the same as the empty chunk ratio? So as the number of FP labels increases we decrease the number of empty chunks we add in addition?